src/hotspot/cpu/x86/macroAssembler_x86.cpp

```diff
@@ -4630,12 +4630,23 @@ void MacroAssembler::verify_heapbase(const char* msg) {
   assert (Universe::heap() != NULL, "java heap should be initialized");
   if (CheckCompressedOops) {
     Label ok;
-    push(rscratch1); // cmpptr trashes rscratch1
-    cmpptr(r12_heapbase, ExternalAddress((address)CompressedOops::ptrs_base_addr()));
+#ifdef _LP64
```

Comment:
Really?  This makes a difference in performance?  If it does, maybe one of the verify_heapbase() calls is in a place where it is called too frequently.